lib/templatesimplifier.cpp

```diff
@@ -726,21 +726,34 @@ void TemplateSimplifier::addInstantiation(Token *token, const std::string &scope
         mTemplateInstantiations.emplace_back(std::move(instantiation));
 }
 
-static void getFunctionArguments(const Token *nameToken, std::vector<const Token *> &args)
+static const Token* getFunctionToken(const Token* nameToken)
 {
-    const Token *argToken;
+    const Token* functionToken = nullptr;
 
     if (nameToken->strAt(1) == "(")
-        argToken = nameToken->tokAt(2);
+        functionToken = nameToken->tokAt(1);
```

Comment:
hmm I suggest this refactored code:

    if (Token::Match(nameToken, "%name% ("))
        return nameToken->next();

    if (Token::Match(nameToken, "%name% <")) {
        const Token* end = nameToken->next()->findClosingBracket();
        if (Token::simpleMatch(end, "> ("))
            return end->next();
    }

    return nullptr;